<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Haochen Ning" />


<title>Introduction to the iimi package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to the iimi package</h1>
<h4 class="author">Haochen Ning</h4>
<h4 class="date">First created on 2023-04-29. Updated on 2024-07-17</h4>


<div id="TOC">
<ul>
<li><a href="#introduction" id="toc-introduction">1. Introduction</a>
<ul>
<li><a href="#installation" id="toc-installation">1.1.
Installation</a></li>
<li><a href="#loading-packages" id="toc-loading-packages">1.2. Loading
packages</a></li>
<li><a href="#pre-processing" id="toc-pre-processing">1.3.
Pre-processing</a></li>
<li><a href="#downloading-example-bam-files" id="toc-downloading-example-bam-files">1.4. Downloading example BAM
files</a></li>
</ul></li>
<li><a href="#converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame" id="toc-converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame">2.
Converting BAM file(s) into coverage profiles and feature-extracted data
frame</a>
<ul>
<li><a href="#state-the-path-to-the-folder-of-your-bam-files" id="toc-state-the-path-to-the-folder-of-your-bam-files">2.1. State the
path to the folder of your BAM files</a></li>
<li><a href="#create-a-data-frame-that-contains-the-coverage-profiles." id="toc-create-a-data-frame-that-contains-the-coverage-profiles.">2.2.
Create a data frame that contains the coverage profiles.</a></li>
</ul></li>
<li><a href="#predicting-the-plant-samples" id="toc-predicting-the-plant-samples">3. Predicting the plant
sample(s)</a>
<ul>
<li><a href="#using-pre-trained-models-and-no-customization" id="toc-using-pre-trained-models-and-no-customization">3.1 Using
pre-trained models and no customization</a></li>
<li><a href="#customizing-your-own-model" id="toc-customizing-your-own-model">3.2. Customizing your own
model</a></li>
<li><a href="#customizing-unreliable-regions" id="toc-customizing-unreliable-regions">3.3. Customizing unreliable
regions</a></li>
</ul></li>
<li><a href="#visualizing-the-coverage-profiles" id="toc-visualizing-the-coverage-profiles">4. Visualizing the coverage
profiles</a></li>
<li><a href="#references" id="toc-references">5. References</a></li>
</ul>
</div>

<div id="introduction" class="section level2">
<h2>1. Introduction</h2>
<p>This vignette aims to give an introduction on how to use the
<code>iimi</code> package for plant virus diagnostics and how to
visualize the coverage profile for the sample mapping. We also included
a tutorial on creating unreliable regions.</p>
<div id="installation" class="section level3">
<h3>1.1. Installation</h3>
<p>First, let’s install necessary packages. You may skip this step if
you have installed the packages before.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install iimi</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">&quot;iimi&quot;</span>, <span class="st">&quot;httr&quot;</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># install Biostrings</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;Biostrings&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-packages" class="section level3">
<h3>1.2. Loading packages</h3>
<p>We will load necessary packages before we start any analysis.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(iimi)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httr)</span></code></pre></div>
</div>
<div id="pre-processing" class="section level3">
<h3>1.3. Pre-processing</h3>
<p>To get started creating coverage profiles and feature-extracted data
frames, we need mapping results. We used Bowtie2 to map the samples
(paired-end or single-end) against the <a href="https://github.com/virtool/ref-plant-viruses">official Virtool
virus data base (ver. 1.4.0)</a>. We recommend Bowtie2 or minimap2 since
we have tried both and they yield similar result with minimap2 having a
slight decrease. We let both software to report all alignments
(<code>-a</code> mode for Bowtie2, <code>--secondary=yes</code> for
minimap2). You can also use other mapping tools. Make sure that you
convert your mapping results to indexed and sorted BAM files using
Samtools.</p>
</div>
<div id="downloading-example-bam-files" class="section level3">
<h3>1.4. Downloading example BAM files</h3>
<p>We provide three example BAM files to demonstrate how to use iimi.
These files are sourced from the dataset used in the VirHunter paper
<span class="citation">(Sukhorukov et al. 2022)</span>, which we also
utilized for external validation in our manuscript <span class="citation">(Ning et al. 2024)</span>. You can download these files
directly from the Recherche Data Gouv website <span class="citation">(Candresse, Marais-Colombel, and Brault 2022)</span>.
We recommend storing all BAM files in a single folder for ease of
access. Here, we provide a short tutorial to guide you through using
iimi functions to make predictions on the real data.</p>
</div>
</div>
<div id="converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame" class="section level2">
<h2>2. Converting BAM file(s) into coverage profiles and
feature-extracted data frame</h2>
<p>Let’s convert the indexed and sorted BAM file(s) into coverage
profiles and feature-extracted data frame.</p>
<p>We will use the coverage profiles to visualize the mapping
information. The feature-extracted data frame will be used in the model
training and testing process.</p>
<p><strong>Note that both training and testing data need to go through
the conversion step.</strong> In our example, we stored the conversion
for both the testing and training datasets in the same object. You can
do the conversion separately for your data.</p>
<p><strong>Important: the example code does not work unless the path to
the folder that stores your BAM files is provided.</strong></p>
<div id="state-the-path-to-the-folder-of-your-bam-files" class="section level3">
<h3>2.1. State the path to the folder of your BAM files</h3>
<p>If you already have coverage profiles in run-length encoding (RLE)
format, go to section 2.2.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>path_to_bamfiles <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;path/to/your/BAM/files/folder&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">&quot;bam$&quot;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">include.dirs =</span> <span class="cn">TRUE</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="create-a-data-frame-that-contains-the-coverage-profiles." class="section level3">
<h3>2.2. Create a data frame that contains the coverage profiles.</h3>
<div id="convert-bam-files-to-a-list-of-rles." class="section level4">
<h4>2.2.1. Convert BAM files to a list of RLEs.</h4>
<p>You may skip this step if you already have converted them to RLE
format.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>example_cov <span class="ot">&lt;-</span> <span class="fu">convert_bam_to_rle</span>(<span class="at">bam_file =</span> <span class="st">&quot;path_to_bamfiles&quot;</span>)</span></code></pre></div>
</div>
<div id="convert-the-list-of-rles-to-a-feature-extracted-data-frame." class="section level4">
<h4>2.2.2. Convert the list of RLEs to a feature-extracted data
frame.</h4>
<p>In this section, you have the option to use the provided unreliable
regions. We recommend to enable the profiling and filtering step as it
eliminates false peaks. Examples will be provided in section 4. If you
wish to disable this mode, simply set
<code>unreliable_region_enabled = FALSE</code>. If you wish to enable
this mode, you do not need extra codes. If you wish to use your own
unreliable regions, please refer to section 3 and input
<code>unreliable_region_df</code> with the unreliable regions that you
customized. If you would like to use the provided mappability profile,
you do not need extra codes. Here, we enabled the mappability profile
and nucleotide filtering mode and uses the provided mappability
profile.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">convert_rle_to_df</span>(<span class="at">covs =</span> example_cov, <span class="at">unreliable_region_df =</span> unreliable_regions)</span></code></pre></div>
</div>
</div>
</div>
<div id="predicting-the-plant-samples" class="section level2">
<h2>3. Predicting the plant sample(s)</h2>
<p>To make predictions, use the converted mapping result of the
sample(s) that you wish to detect as the input, <code>newdata</code>.
Make sure you have converted the indexed and sorted BAM files into
feature-extracted data frame from the section above.</p>
<p>After preparing your test sample, you can choose to test the data
using our provided training model or the model you trained using
<code>train_iimi()</code>. The tutorial of training your own model is
provided in the next section.</p>
<p><strong>Note: if you wish to customize unreliable regions, please go
to 3.3.</strong></p>
<div id="using-pre-trained-models-and-no-customization" class="section level3">
<h3>3.1 Using pre-trained models and no customization</h3>
<p>If you wish to use provided training model, only input your data to
<code>newdata</code> and choose a method of your wish using
<code>predict_iimi()</code>.</p>
<p>There are three methods that you may choose from: <code>xgb</code>,
<code>en</code>, and <code>rf</code>, which stand for pre-trained
XGBoost, elastic net, and random forest models. The example below uses
the pre-trained XGBoost model.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>prediction_default <span class="ot">&lt;-</span> <span class="fu">predict_iimi</span>(<span class="at">newdata =</span> df, <span class="at">method =</span> <span class="st">&quot;xgb&quot;</span>)</span></code></pre></div>
<p>The detection of your plant sample(s) is finished. The prediction is
<code>TRUE</code> if virus infected the sample, <code>FALSE</code> if
virus did not infect the sample.</p>
</div>
<div id="customizing-your-own-model" class="section level3">
<h3>3.2. Customizing your own model</h3>
<p>If you would like to train your own model and use this model to test
your data, you can use the codes below to train a new model with your
own data.</p>
<p>Ideally, the number of the samples used to train the model should be
bigger than 100. However, we are only providing a tutorial on how to use
the <code>train_iimi()</code> function, only two samples are used to
train the model since <code>example_cov()</code> only contains three
in-house data’s coverage information.</p>
<p>First, we need to prepare our training data. We are using a 80/20
random split to split the three samples. This means that two samples are
used as the training data, and one sample is used as the testing data.
If you are training your own data, treat the training data as your data
that you want to train the model on; treat the testing data as your data
that you would like to have a prediction on.</p>
<p>If you have separate training and testing data, you may only need to
convert your data from mapping results (BAM files) to fetaure-extracted
data frame and name your data as the following variables.</p>
<p>Here are some definitions/explanation of the objects to input in
<code>train_iimi()</code>:</p>
<ol style="list-style-type: decimal">
<li><p><code>train_x</code>: the feature-extracted data frame of plant
samples that you would like to train iimi model on. Make sure that you
have mapped the samples to the virus database and converted the mapping
result to sorted and indexes BAM files.</p></li>
<li><p><code>train_y</code>: the known truth or labels for your
<code>train_x</code> data. Please label the data to make sure that it
has a detection label for virus segments as well.</p></li>
<li><p><code>test_x</code>: the feature-extracted data frame of plant
samples that you would like to predict using your trained iimi model.
Make sure that you have mapped the samples to the virus database and
converted the mapping result to sorted and indexes BAM files.</p></li>
</ol>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set seed</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># spliting into 80-20 train and test data set with the three plant samples</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>train_names <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">levels</span>(<span class="fu">as.factor</span>(df<span class="sc">$</span>sample_id)),</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">length</span>(<span class="fu">unique</span>(df<span class="sc">$</span>sample_id)) <span class="sc">*</span> <span class="fl">0.8</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># trian data</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># train_x is the feature-extracted data frame of your train data</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>train_x <span class="ot">=</span> df[df<span class="sc">$</span>sample_id <span class="sc">%in%</span> train_names,]</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="co"># train_y is the known truth or labels for your train_x data, indicating the presence of specific viruses in the samples</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>train_y <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(train_x)) {</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  train_y <span class="ot">=</span> <span class="fu">append</span>(train_y, example_diag[train_x<span class="sc">$</span>seg_id[ii],</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>                                         train_x<span class="sc">$</span>sample_id[ii]])</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a><span class="co"># test data</span></span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a><span class="co"># test_x is the feature-extracted data frame of the data you would like to predict</span></span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a><span class="co"># here we used the sample that is not in the training set for demonstration purpose</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>test_x <span class="ot">=</span> df[df<span class="sc">$</span>sample_id <span class="sc">%in%</span> train_names <span class="sc">==</span> F,]</span></code></pre></div>
<p>Then, we plug in the variables into the <code>train_iimi</code>
function with the default XGBoost model.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">train_iimi</span>(<span class="at">train_x =</span> train_x, <span class="at">train_y =</span> train_y)</span></code></pre></div>
<p>Now, we have a trained model using the toy data.</p>
<p>Then, the process to detect which viruses infect the plant sample(s)
is the same as previously described, except we are using a customized
trained model.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>prediction_customized <span class="ot">&lt;-</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">predict_iimi</span>(<span class="at">newdata =</span> test_x,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">trained_model =</span> fit)</span></code></pre></div>
<p>The detection of the plant sample(s) is finished. The interpretation
is the same as above.</p>
</div>
<div id="customizing-unreliable-regions" class="section level3">
<h3>3.3. Customizing unreliable regions</h3>
<p><strong>Note: if you would like to create your own unreliable
regions, please customize them first, then extract features to build a
data frame from section 2.2.2. using customized unreliable
regions.</strong></p>
<p>If you would like to create your own mappability profile and high
nucleotide content regions besides from using your own training model,
you may use <code>create_mappability_profile()</code> and
<code>high_nucleotide_regions()</code>. Both functions’ output is a data
frame with the start and end position of the unmappable region, the
virus that the region is on, and the category that it belongs to.</p>
<p>Mappability profile is a profile of areas on a virus genome that can
be mapped to (1) other viruses or (2) host genome. We choose Arabidopsis
Thaliana as our host genome.</p>
<p>High nucleotide content regions is a profile of areas on a virus
genome that has (1) high GC content and/or (2) high A nucleotide
percentage.</p>
<p>Including these two profiles into iimi ensures that there are no
false peaks like the ones described in the previous section.</p>
<div id="mappability-profile" class="section level4">
<h4>3.3.1. Mappability profile</h4>
<p>Here is a short tutorial to make mappability profile.</p>
<p>First, split each of the virus segment from the virus database into a
sliding window series with window size of your choice and with step size
1. The default value for window size is 75. You may choose any window
size you want.</p>
<p>Then, map one virus segment with each other, until you finish mapping
it to all virus segments in the virus database. Also map the virus
segment with a host genome of your choice. We chose to use Arabidopsis
Thaliana.</p>
<p>After mapping, sort and index the resulted BAM files from the mapping
step.</p>
<p>Next, it is time to assemble the mappability profile:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if you would like to keep unmappable regions that can be mapped to other viruses or the host genome separate into two data frames, you may use the following code:</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="co"># input your own path that you would want to store regions on a virus that can be mapped to another virus</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># you can customize the name of this type of mappability profile</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>mappability_profile_virus <span class="ot">&lt;-</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/virus&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region (virus)&quot;</span>)</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># input your own path that you would want to store regions on a virus that can be mapped to the host genome</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="co"># you can customize the name of this type of mappability profile</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>mappability_profile_host <span class="ot">&lt;-</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/host&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region (host)&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if you would like to keep everything in the same data frame, you may use the following code:</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>mappability_profile <span class="ot">&lt;-</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/of/both/types/&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region&quot;</span>)</span></code></pre></div>
</div>
<div id="high-nucleotide-content-regions" class="section level4">
<h4>3.3.2. High nucleotide content regions</h4>
<p>Creating the high nucleotide content regions is much easier than the
mappability profile. We only need to use
<code>create_high_nucleotide_content()</code> function.</p>
<p>Here is an example:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>high_nucleotide_regions <span class="ot">&lt;-</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_high_nucleotide_content</span>(<span class="at">gc =</span> <span class="fl">0.6</span>, <span class="at">a =</span> <span class="fl">0.45</span>)</span></code></pre></div>
<p>The default threshold for GC content is 60% and is 45% for A%. The
thresholds are customizable.</p>
<p>You have created the mappability profile and the high nucleotide
regions and can use them to convert your training and testing data to
feature-extracted data frames. Please refer to section 2.2.2. to see how
to do so.</p>
</div>
</div>
</div>
<div id="visualizing-the-coverage-profiles" class="section level2">
<h2>4. Visualizing the coverage profiles</h2>
<p>Next, we can visualize the coverage profile by using the
<code>plot_cov()</code> function.</p>
<ul>
<li><code>plot_cov()</code>: plots the coverage profile of the plant
sample and the percentage of A nucleotides and GC content for a sliding
window of k-mer with the step as
<ol style="list-style-type: decimal">
<li>We used the default setting of k = 75.</li>
</ol></li>
</ul>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>oldpar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot all segments of one sample, you can try:</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cov(covs = example_cov[&quot;S1&quot;])</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot all segments from all samples, you can try:</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cov(covs = example_cov)</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot certain segments from one sample, you can try:</span></span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>segs <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;42jtlrir&quot;</span>, <span class="st">&quot;m0kacxse&quot;</span>)</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>covs_selected <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>covs_selected<span class="sc">$</span><span class="st">`</span><span class="at">305S</span><span class="st">`</span> <span class="ot">&lt;-</span></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  example_cov<span class="sc">$</span><span class="st">`</span><span class="at">305S</span><span class="st">`</span>[segs]</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a><span class="do">## if you have many segments that you would want to plot, you can try the following code with the numbers changed</span></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="do">## to find the index of your desired segments:</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>covs_selected<span class="sc">$</span>S1 <span class="ot">&lt;-</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>  example_cov<span class="sc">$</span>S1[<span class="fu">names</span>(example_cov<span class="sc">$</span>S1)[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">72</span>)]]</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">4</span>), <span class="at">nrow =</span> <span class="dv">3</span>))</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_cov</span>(<span class="at">covs =</span> covs_selected)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAqAAAAHgCAMAAABNUi8GAAABEVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZmYAZpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZrY6kJA6kLY6kNtmAABmADpmAGZmOgBmOjpmOmZmOpBmZjpmZmZmZpBmZrZmkGZmkJBmkLZmkNtmtrZmtttmtv9paWmQOgCQOjqQOmaQZjqQZmaQZpCQkDqQkGaQkLaQkNuQtraQttuQtv+Q27aQ29uQ2/+2ZgC2Zjq2Zma2kDq2kGa2tma2tpC2tra2ttu22/+2/7a2/9u2///bkDrbkGbbkJDbtmbbtpDbtrbbttvb25Db27bb29vb/7bb/9vb////AAD/tmb/tpD/25D/27b/29v//7b//9v///9o9graAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO2dC3vktnWGobU3ley429RaO/Ft62wiru1N0sYXre2mqRPJ3rhuG2UiV+L//yHlDXeAPCAPOSD5fc/uDIcDHBwcvDwAKGlGlBCUscSxHYCgPgFQKGsBUChrZQjoj/+2hyYXld+/1fR4IUBfPrz58rI+uPv5q1e3ZyeXd4/FeXl79upV8/a1ECeXsuz1uT6qjw+npX3CKxc7IRu8FheyiHijKvRCiNrm7WuXdz8/ubh7+6qzGzSQqWrPR4RUn7HOOQVq4+3jtfjg7NXLevzqIn/vjpvnTnXTMwZuGUC/ff/RTcvZ9ZsvLl5UwWleXZ++qPFo4nN9Xve0+v/FsyqwL944eV+c11E8nP30tLz/6PN3f2gC0RT79OM/vfXFx789v35F/Ppw9sZpY+DFRVVYPPhtZbq2VEobb7741eOT95qSJ7+vj6rY339UvX//rDZ4OL17+8/PTt7r6ooHV4tEZLJqz5NCWh8fxMkbn7/7tRC/rnvchikUutZY9fiTxyc/vTic1+N3XQXuRXdcP8sAv2iG6cuLKryfNbC++aVs93KgDxQtlEFvH91U/+qjw6vfVN07vRYVJ3ef/OM3zdvN5V7ntfp/fTm2Aa97XheuCnz4s/at5sTDrz/8xa/e/bqOzGn7/u2jlw+umrA2A/fqpzfSRttgXfLHX1TRvOgK1Y2+WWFaodpl0K7uMvFgUOV5WkgvGqKrSFbgPvpahTYUuvYKrh+rMmfVdFO11LzbHTfPXYDbpk/LNryH0/tnD29qK900NVVLA9p2SQHRzQ3VU3u6fatsr2kD0Ns3PmvfagF9efbhWZtfO0DLL396Xkqbp4d/+MSw0VquHp9+3QHaJVAhHv61OlonoHUfEkPaAFpF0gY0FDrFZxPEl+cSUH38UgW4sf+Td67a8FYT0s8fXNVW6vMM/VwO0EMTpttuIqifm1mnnY+qy/2iXj7V/1+eXBqANlN8ef9x81Y3xX9299bnj7sheSHqKb48VJXahFm9ribvbmSaOeuzzlq1/qyOXmlWnfUAVgNQNVxN8VWT760N0NrzpJA2U/xpHck6BC/VFO+HrjVehbfOv42NBspXTtRxOxYqwPVK6Wd1eG+7ZVhtRa2Gp2m5XfyXE1Yk9bQReauKyUXo/LfBs4H3b89OY7Zz14iQ1pGkZTZyAOdVhreZIEhrBKACcjRxCCBHUwGdNh7b01RAebzYjgAoswAorwAoswAorwAoswAorwAoswAorwAoswAorwAoswAorwAoswAorwAoswAorwAoswAorwAocxeWAnQDgScJgALQrAVAAWjWAqAANGsBUACatQAoAM1aABSAZi0Auk5AxdRf3V+LACgAzVoAFIBmLQC6UkDrkhsI/qAA6GoB3UcSBaArBnQL0R8SAF0loHXyBKDJ9dcp3plyOUC7LLpxAdD1AsrQXP6iA3poPojE/RzSDQRolYAyNZe/yIDeP2s+wKz7mFRSlZUIgOYsMqDN50yqJ1KVlQiA5qzkDPo6AO23tkh1AOrr9gxrUIq1RaoD0FH11ykAmrOwiwegWQu7+HUDuoUB6BV28cw/jiHbmjQjAVBPG97Fs3aCamvajARAfW13F38UQKfNSACUUNP/EoZ16rgZdNyMBEA9NV/IJoT7HVobiM9xAJ02IwFQTxWg7beNbm8XfxxAJ1UHoJ6ab1nd5i7+OIBiF08SHdDHD394Wj29iww6YI0mtl38BkagTwmbpOZLdF/DLn7IGk1cu/gtjECf8LP4le/itzACfQKgK9/Fb2EE+gRAc9nFJ91XBqCe7h63AXyAXfyANZqm3VcGoL7u3roMnN1AeI4G6IT7ygA0oIN7tQ9XWYWOBuiE+8oAdFT9depIgE66rwxAR9Vfp461SZpyXxmAjqq/TmWyi0+qDkBH1V+nAGjOAqDrA1QA0JH116ljbZJG31cGoGPrr1NHyqDj7ysD0LH116ljTfGj7ysD0LH116m1rUGdz/7ewAj0CYAC0KwFQAFo1gKgADRrAdC1ASoA6Oj669TaAC0B6Nj669TqAC0B6Mj669S6ABUAdHz9dWplgJYAdHT9dQqA5iwAui5ARfsAQEfVX6cAaM4CoM7vXjBYm7M6AJ1Sf51aE6ACgE6pv06tCtDuEYCOqr9OAdCcBUDXCKj5EU4b/4BQALp2QDdOKABdIaClA+gGBiEqALomQOW7AHRc/XVq7YBue5IHoAA0awHQDQC6hVGICYBuAtANDENEAHRFgGoOPUA3MA5hAdA1ARotBkBJ9depNQLqvcPbiYwEQDcBaP3j+W1mUQC6EUDp37G0LgHQbQDq/LH8dgRA1wPo0C2oDYyFLwC6FUA3MRa+ACgAzVoAFIBmLQC6GkAHd+kbGAtfAHQ9gM7cdJ4CoAA0awHQDQG6gcHwBEC3A+gWBsMTAAWgWQuArgVQwgS+gcHwBEBXA+jsbWcpAApAsxYABaBZC4AC0KwFQAFo1gKgAHRSzblFB/TQ/FXByWVP/XVqQ4CObTzjPxchA3r/7KJ+un10E6+/Tm0H0NGNbwHQu7evjCdSlZUIgG4CUJlBXwegQ9bmqT4roPn+ogl9DXp7tsE1qCgBaLkRQAn11ycA2s3vGwB0k7v4IwKaGs95Ac12IHe+iz8eoMnxnBvQTEdy57v44wGaHE8Ausdd/PEzKDmeMwOaK6HYxR9rDZoaz7kBzXQox+/ihdjAJ6phF68r5TmU2MVjFz+l8uzCLh67ePmc5VhiF7/3XXz0mxnyEHbxO9/FG3kzy7HELn7nu/jtAEqovz5ltIsfuisyF6CTas8vALrzXbxZJcdtEgDd0i5+REdsQPMbTvou/nE7BT3AJmnQIkWz7OInAprjcNIz6N1bl4Gz+fUoSQpQtn4cdRe/a0DLw8Mb/2R+PUrS8QBN3cXT7E4FNL/xxBr0WIAmVgeg7osR9dcnAApAsxYABaBZSwLKd3+FvItPvCsCQN0XI+qvT8cDNPWuyLoA5cMCgB4L0MS7IisDlO0HVAD0aICmVQeg7osR9dcnADo/oMHgkgMOQPcNqNtxnjgIAqDElgDoOgAdM5xjKrAEQji/ZQpARwuATrYQsClK+w9JQoCO6tGOAWXrCABtTQBQFgHQ6SbCFgAoiwDodBNhC8OAjtj2AVAeizNUXzWg4VYAKEESUL6OAFAbUJtWswztg5MAKACdaCJsYRhQ2r0TANp1IW9AR901HFcBgOaklQA67geD4ypMDoQdzzCgskPDjQHQVQA6W+vHAVQAUKIA6HQTYQMi8EIXERa/w9aopQfqr08AdLqJsIE+QAUApWodgNJtcgA6NRIEQEsAShQA9c9wAhr++HsAStYqAE1AhgHQyYQK7xGAjtY6AJ2x9YkXRNxkYC9vlgCgNAHQ0Llpvx5LArTXgVhZAMpkkbd6ikUWQCcSak7rAHSCRAlAoxUA6PElBABdDNA4ZwMtAdDcAU2abfMC1Fgp2H8on9ASAM0e0NlaFz2/pz2+GwKAMmnvgIpZADXvcJp/fBxfd/Y3tV9Axe4Brf/xA6oPLCYBaKpEuW9A203MUoBav50MQCkCoH3lGQB1MASgiQKgfeXZATVNAlCKACiToVhFADpNKwE07YeOOQEafSdO7pA9AMphlrd6mr2jA9pTD4AmSQBQJkPUegA0SaIEoDyGqPUAaJIAKJchaj0AmiQAymWIWk/FGoBSBEC5DFHrCXlLAoBStHNAh35nfg5Au7+V67mDP2QQgLLYZa1+JEDH9qOvWvuXJH7LADSk3QPKZYlerU2hgUujt9JEpwCob5e1+lyAsllKqhX+czwA6qv9ZVABQCdbSqwVKgFAfbVkAtDpljhqDdycmmIegAYMc1YfD2hvv7ICdODm1CTzADRgmLN6oj2juP5tA0o7E9udUKm/HgDdKqBCbppp7Uxsd0Kl/ooANG9AU80ZgDb9i+yax21m5qnUXxGAbhfQ7g+LhVuA9NFLywJK/Os9AMpjmLH6REDL0v3uYSHK6Lw/peHxlfqr7hRQsQtA1cEaAKXs5nrtH5oFzcklm0vHlGgf9gGoEC6gIxxJbJ2tLhnQ+2cX9dPtoxs2l46o7QNqfSySCNOa4kha62PqEm439Nm/e/vKeGJx6YjaAaDmOf2BSeO+Bju5db7KyRn09Q0BKo4JKG3JxAGofhXY1Q8aGtU6X2X6GvT2bDtrUBEAlKknVDPEJRMzoCMMjWp9VOVg7X3u4iWT5g8DFwaUuGQaC6gInp4L0C6KLF8P0ndyN7t4NeMdD1DakindKRGcxxcCNKUOscl97uIzAJS2ZBoBqAhu1Ft+ZgJUNnlUQL0pKRKJVSgHQEnVxwEamGvnBFRkAeimdvE5AEpaMo0BNOpWUjqhF1W/lcLzHYp9J/ezi++eLEB5usK7ix+zBo2enun7QtS+7MiAJhjNXiFAmboyehcfTEK8gDLYCRQU/tE4TV+DXgvx0L7gAWjE9KDm2sVHNCOgyb4kWUoB9HC6tV18eTxA59rFRyRSE9waAb0+38rP4nMAlFSdFdApjjAUHGeJDujjhz88rZ7eXWkGjXxRz1HXoMNLJkZAE1eIqwO0npNOLm9fW+suPkNACUsmTkBTK3AXHGdpN7v4DAElLJkAKAA9FqCkJRMfoKmWAOjCyg1Q2pJpV4AGTQHQnezikwElVuAEAICqF+YRAKV4MrHYaFsAFICSPJlYbLQtALoPQNMFQJcVAE0VrW0AyqRhQHn6AkB5WwSgOQN61OgC0EUFQFMFQBcVAE0VAF1UADRVAHRRAdBUAdBFRQCUpTMAlLdFAApAIwKgiyoJUDEeWgDK2yIABaARAdBFBUBTRWqc9bOPAKh6YR4B0LBogM7dIgANAKo/4CP9A4e2AyipdQDKpSRAVel9A0poHoDySIwFNHmRtQtAee/M9RoDoEFASwDa0/yEJXpyg/sA1P1M135AhQnobJ/IQameLaB6hpm7QQAaALRUWIoytYs7AFQItY0EoDxyP3kzAmhpUAlAI813n1HSTjPcDvr2dgmofQhAowruWvSXTAFQLokyAdBuANQJAGqdmnCPeEyDuwHUT5Ty0AO0tB9n+uBXSvVjR3doTbhAgwA0DqiwTpCbmqTcAZ35e4f2C2hgqSkPY4AKACpva+j4ze2RZ38XgAoAOk5dKNR+aP7vbdspoOrBeK0OewCVG9bUtsYrQ0Dby7v9KjkAOouSABUa0DG3ozcH6Ky79mCL8RMAVL+ZA6BHD67zEzgAOo+MG5secH2Aqnd2DOjSXxe8T0D1YxqgtoGUxkYrL0AXSZp2e70nAGgZBDSllxsDdGkBUP0oDwFoRto9oIH9UgRQs2cAdCkBUH1Cng0DWgLQYwiA6hPybARQ30JCY6MFQPtObBNQYTwB0MwFQI0z3SEAzUgA1DjTHQLQjARAjTPdIQDNSDsEVADQFWmPgJrPADRz7Q9QAUDXpB0Cah0MAxr87QgAupQAqHUuBGivkYT2xgmA9p0AoAD0yAKg1jkPychnZQDQpQRArXNkQOndBKDT5PYZgNqARlAEoEsJgJb9gA5Yobc3TgC07zUABaDHFgAtAWjOAqAlAM1ZALRcDaDZx3YOAdByFKDkfgLQaXIjDUBpIQCgSwmA9gI6aIat4HD17GM7i3YGqHABFSUAzVp7A9Q+6j7m0nofgGYl54d5ewPURRKA5qf4h7r0RuTQfBbfyaVjjM2teTQXoKLTYMGohuOZe2xn0yhA759d1E+3j26itnJUANC/uUo01LwSobPhM2ENxnPpT+bMSKMAvXv7yngqVQ7RlfOU419ZenwSAQ2aDTRDskaJJ9HQBhUNAyWDvn5lnd5xGMNKzaCIZ7/oa9Dbs/oCX9sadHGRA4J4krT1XfziYtzFQyUAZRcA5RUAZRYA5RUAZRYA5RUAZRYA5RUAZRYA5RUAZRYA5dVkQCFHE8cDcjQR0HGV5yqbhxM84m6SvQtHcBCAsjjBIwDK3GQWbGThBI8AKHOTWbCRhRM8AqDMTWbBRhZO8AiAMjeZBRtZOMEjADp/kxDEKgAKZS0ACmUtAAplLQAKZS0ACmUtAAplLQAKZS0ACmWtCYAehLgYKtN8ikZb0HwM6P6ZEKfEsmX5wivW5831ObVsZffBFdkur/ha8nsxSfRBTLBH9XE8oHdvXbqfkeGXeVw50RY0H0NFr0/rT96glS0Pp3UviYXL27NzohP3H12qng3b5RVfS34vpjlGH0S6PbKP4wG9fXTTthLX/cd/qi6WtqD5GCt/fU4vWwFKLHz/0XvnRCfaONGd4BRfS34vplhLHUSiPaqP4wGtslj5YijT19m8LWg+xguTy15XkwKx8PV5NcXTyh5eOavWGWQnWMXXkt+LafaSBpFqj+pjRoBWmT4lDnaxeOHq+iQDWv7fTXlNtMstxpa8XkwzNwegZB9nneJbX4izw+1rl0mzK3U9cF3/GVbC2iFlncEp3pbsXkyzlTCIZHtkH+fdJDW+0NbX7Wn6Jqm65sgL92vqJqm2e023yyq+lvxeTPSMPIh0e2Qfs7nN9KJJdAm3mci3jhJvMyXY5RXrbaZzRv/nus1E8hE36qGsBUChrAVAoawFQKGsBUChrAVAoawFQKGsBUChrAVAoawFQKGsBUChrAVAoawFQKGsBUChrIVv+WDQxCGAHE0FdNp4bE/4niReAVBmAVBeAVBmAVBeAVBmAVBejQc0sIaFJhCGeAaFDMosZFBeAVBmAVBecQH6N1PtqW8vy9un8ZrfXbX//bNOC/5NsfvnHz5/Wj284/yB/1++8Y/iKgr7udbfH91Ey1NssgGKeEp3oy9o6qr8LRbQ73/3wVX1/w8fXP34vDm6qf/Xx9/9ywc3313dP39SdeH7519Uzz8+//DT72/unjdlZGBDd20ry+V/VQ/3/1mWd1XFrsqfv3zzr9Wr9igemE5FF8jCDOhXf6jHqHaxduz7559Xj/fK5n8/r7vx4aet070BGSvEMxyQwAuaeqo0Af328vazby8PF7ef/c/NX6qj5sxTefzd1Vdf/PC0Llo/f3XVnLlqa/U0+lU9Os+v5PE3skprpj1K70qluyd3TxrPb5+2lhrrymbtdt1W63RyQChCPPsCwgzoRVkH6Paz767+Xvnz9S//ozq6e1L/18df/f6Pn9YTUfX8+9/cVOX//Jub+nxvQOtwffW7y/LuX+uA3v9VVqkCUJlrj9K7Uhv+5R+fNzPl3ZPWUmX9Rtus3P60drJ1OjkgFCGefQHhBfTunfffudIB/fdffXL+7T8/v6z/y+Pvrg5PPrmoe1A/H/7pvfOq7C/r870BvX/+/vtP/rddM9XFZZXDz15WZtqjwSkpZPc3N+WhTkCVa4fGUm35oGz+rnK7drJ1OjkgFCGefQGZfRcfXshL/eXq5UKffDSguIvDTi66i99bPHGbabpwm4lXAJRZAJRXAJRZAJRXAJRZAJRXAJRZAJRXAJRZAJRXAJRZAJRXdEBvz04+f1x/Y128PpQQEMSTJDKg9x/XX8BQ3r1r/0gBAXVEDQjiSRMZ0PrrMOtvIHnbvuQRUEfUgCCeNKVm0NtHuOJ7lZhBEc8B0degB4E1E0HkgCCeJGEXzyzs4nmVCqheM+GvEINKDAjiOSBkUGYhg/IKgDILgPIqZZNU68T53X8E1FHCJgnxJIh+m+lZ84vPuC0yIPJtJsSTpJQb9cYTqcoulXCj3nhKrr4bJWfQ1xHQXqVmUMSzXym/LII1E0EJvyyCeBKEXTyzsIvnFQBlFgDlFQBlFgDlFQBlFgDlFQBlFgDlFQBlFgDlVWaArn98ACivACizACivACizACiv8gI09gu7K/pFXgDKq1UAuqbfNAegvMoM0LA9ALpf7QvQBYYfgPIqK0AFBdAJLS6RhwEor/YE6CILBQDKq/GAxv5MdlqGCzKUAmgPhMqMZyzR59ZMZEOXZso0ij87Dog/g06IsYgsN0VrtXnHfd8p3TPECn9dRgKbA6As1ben3AANEiZKBam7j3JLA9CNaRZAxzLaC6igAhprnANQWV9EawFQXs0BaDd+ifXKIKDN1N4dBAD1cIztsyxTMiVbgFI6L4QwOhheLxPM9LUwrfr2tDlA4w0zAhrN9QCUW0cDNDa4zvxrACrNmlW9GX0IUKGSrEG7ED3VzK40zbX/lUOxsuMEQEs7COyAdiPYg6AsFntX0kIF1MiKHqDusQa0g1LO2X0LUb0nUl+zrhjtmO0JSJoAqJN0MgRUQrYcoKW6RxDuYxhQ1VkAyqpjAmpM36FJVag35WQvzK23XDnoevretjAoMttzuqrh6sDyAXW9MvojDECli0K7GA5ImgCos3biB9RYqbmFhAuo8OpKD8tS4yBTpNA0KSNGWYmS2bB3LVqACg2oiACqu6Jn9ZiL4YCk6aiA8jYen5IG682aQWUb0wEt7dE3050yopYDGlA7w7rWNeUm8GRAZTEH0J6ApOmYgI4litmciL9iAzSSptXJjg1h90EhItFxAdWMynImoMIAVFsKeBYEVLtmu2RO6hFAy0UB7VscTWybk1DnupVnByr5ZWYFVJNU6rPdyk8NvK5nFpcIGRQrQE0kyzYRW6nMzIY6SCaAxmytCDf2QkZvDMkChjmnD8GApKm/unOVGOeT2rDcNiMUnEQiJgbed2ZqeX7QMbcvCwOq3+gBVL90ByMOqOyeUco6CgJqtNsHqMy2/qD4m/hFAPXwSALUuOSsumoECNYIgJahzlAA7akxFVBzBjQncGNjoUJjRkkYgBim+gA1E6a0Qge0tBKiAlQ4kdcXUR6AusncfCPMlTCOJIDWrKB7LIz3B31SdaOOBiraPur6Zgrvb3wwuPcfOR9naY+NjpycyKVbCiATUKOfXtCFvqJtQKV13RWNuXFCeUEHtDQ8Mi1b8Lvy3kgCtD+efluq+/ZIqzkp5J7srJDXvPrphLVCUTG0I9A14Nk1jAYdtZ6lK24HDBJKzUdPQPqic/e47c6D+CcCm8nEuj6My83MfKWJrvBcCSYEA1Cz58abVio1L0/5pnTVbkioYxdQu3zIqfiruAjxdNsR4RcyNw4DqhODfltBI22ZEZDj5hq2qO/x1AVU5ehStqP9jtmKvnB1e3bef8VblCn69GUxAKhwXAnOWMak3g+oblC6EwPU9MYenlkBHY6n1054rFQaMmJivanypJVJ9dtWnITxQie1QFozLPuO6vpmBRVjxYXMNX2wR1/4uj75ggyodNTohTnP23VCgAb9NQC1T3vFjAaL0uTdSwYe6uoiN/0sfGfCIgM6GE/by8CFIGFSec68nGQvrFD42VBY5dX1q15Itp2GddkQoCbgzoEBtrqwjCQW6Hn0RUD3z9yPVKcBKrOlLG/mqNilWAQdDic0HzrdYFHYgMZMqh4sB+hAPB0ng0lMZn67ukLKjrYHedS8QWvbhjViMi6FM5h+H4KA6hnOPBF2LBHQ/vohQIUGVCQCWnffw4IOqIxjw2dfCHxA9Uyo3irIoRkRQ0L1AKDm3GSWK/U0mgBoKKDCB7Qwk10bWt+meb0INYEbPZkT0PiXn4bytJHJnVnC7oJ9kTaqe18ULqHCeuqRdq62oT3tzYRd3OpRKOzOzQZoz5fJWsH1PVATpH1SFIV8JzDs0e67ZuSD7Us1IIUR1kIX1g24rHd5yznptrxABtU3lhxHfECdMv4Cq2wCUAzw1OuXilMHqDTbX6ktEwB0lim+v7rRvgh44OROWbAq502rnejx9G13VNb460ulcHxuxiy0KA1dXEQtAGg5DtBmfk53yLJatllYEAEtu9wgJ3kj8scBVC3hGjSckhFAyzigDaGknjSJ2D4hRMenDIZKIWrc63dCe4dlAB368tMi1LCMhjXadoyEWU4dFOEpniwhL/SiAa5NLYMWFaANpDIFJ/lBDv3gl8mKbnlWZ6X6enUBDZst+vdzZECbux9m2cIIaTvZd49FURjXf33shouwQCpijpEBHfzyUzqggeH2AW25mJBCVVREkZYKi6Jt2AKUPYMOf5lsO/t0gBJVu93nalFSp6UAoIW0YA1XIZN2tx1tkHVaHb4J0gAaKkUGlPTlp6MBDdQKUqHPDFkxATWu4GHSik7qum9jRyWUytJgPIUEtFnWeWSF/WnWM5G1ppwJZGf6e1RDpbNFs/spjBeOIzpvCntgCkr0uirBIU3OoP1ffmonVLNFY8/i+1GoDN9WkyW1KfdoEFA5HRWlkN0vy6FYdcGXE5c6k7BdS82g0XjqJYoeQOVkKZ0svSDVgAav7cI5HIifEavC3LQ7tjpXCnXDUPomrQwtKgpFfDtV9a1keoM79OWnjiOxFV8RymIOoKUdkyCgw1elfEoEVPkuTFzjGcsReTYejmf3pEfNBbQIASoruZ66hYbj4I5JvHLRJFxzX6mcDLQcaM9Y2Y0GNCyjSuF33zxlRNJbo+hQF4qIMpC3rCxAA7Qszf3vUCrUbhdqzdV5F5w1JwAaVuB2m9BLOCNpdlNNYcGkB9rzLA6oe/EVsl8arsHpo7nJ4N95Vcv3/lW8/Z5dkBFQsyMBh2LXY9HxoCeeLmUYEVRV6IAavRbWHatA+rZcVDlcbfsL66Kys3jhdocP0KILhn871pgJpcs+RPagOwwYofZWMIUdEQv8uAoHzkDejExCPVmo5M2gJgUddP6k7I1ooZgsVD27kJXJrJNxmU0L1YxTxoVWrbv0oCg4w4DPDmgAO3fBUegsZReN56VCcd3NVtHQBi7BoAp3hxzYyweN+CfnA9S+Cl1ACxUNj2WFswqyla4KXVC/URjnA36ZBiLbT7X3MNarxkAZ1pV7pWeidHckzFN8YB7yTqk056ZJ86IKJtg2JO4MYacPSvoMAOrX6xumeE1eQAvz2A+Xmrk9QEsDF9NDY4tnzTWFB6ibPsxnC1B3XWzsQZw7nsquZcTJL05i5wU0kF76klkkZQUnDzU/yImhyVgAAAQdSURBVLAGL8CecwQNIFr4Wd+vN9satHQXksYcbqUjvSXS66HCMiQ5Ko0oWtsl41HbdaZC5YubKqxE4+R8s0/SRwdw91JknOL7dxameosFrciYFfql3d2EBqI14msOFfmIb/osJ6DONGgBKlONXuZrF409o3ffrNCYBZdJknmDfvl+YcW/0ETKxKwat52x+ucAavZBOmiHmXsXT1JvViVswo1pbWSyJPrkpOlIY6bHzICaF6STa+zMqdKemRcDKwNrftbx01sZM+/JJnWqLGwbqjG1aCv0RRHLM1YHy8K5DErnkjoKoBEV4V7FShuPHI3HThZqrqS0xQuo2aQ5M0t2zOnVmt1t/22T3txtoGZvm5zVbOFalp6UKl0ot4pQuLxrpSgKH1C7Yk6Alv4err80bUM0SXKvfBRAtROlCaj1UpUoSi969vq0OxVYqJirfzMpF1butkJg78CcooW5HLDbMVcVOuOWZTS4WQFaEinoys6Pp7HLJVaYB1BjBO0kaQCj52G7XukC6kfOXCgUan520nNpjI5KrdIjvUJwSgY6Ylo11hueN63yAjSFT8bpva8JZ/oZEjOgPnXWstACtHCLBmWkRGXAWDZIPPWD3YzZmmnFYZqyBjW2eHqd4BYCoNRGjgeoM23HPCnMQXZt+IUDDUg81bpSztteEEJ3LwmAuj1xr6+Qp3kBun7NMMV72/fQUPqrTeut2Cu/npFSDSydTO4uETyPwkt2uyPOceS6A6C8mgNQf+nms9izzwgBau1VrARpoNjBG1jyhtaw9vQd3CHYdyUix44AKK/GAxr+w1YSdP1l+4s6Px2SRwaKQ2vJSPoerua+G8R0NGGRPxTeu7gzKGna7js5pDCB6VYJfA45EqyCDMqrGQD115uxmzjpzQ0DSr3B5t1bTXcEgC4g9jVoeNj4bmAMAZjyc5M5PAGgvOIHdLwvLFrgpyGypeBZAMqrWX6S5IoVmmNfAUrIoEuIH9DQbhiAUgVAHS2zBs0GKk5hil9C68ugmQuA8mqZNWi63dUKgPIKgDILgPIKgDILgPJqEUD3JADKKwDKLADKKwDKLADKKwDKLADKKwDKLADKKzqggx/6D9UiBwTxJIkM6PCH/kO1qAFBPGkiA0r6EgWIHBDEk6bkDNr/JQpQagZFPPtFX4MOfeg/1IgcEMSTJOzimYVdPK9SAe35dl6oVmJAEM8BIYMyCxmUVwCUWQCUV7hRzyzcqOfV5Bv1kCNi4BFPoqiARm4sQyOFeI7QiBv10EghniM04kY9NFaIZ7qwQoeyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWyFgCFshYAhbIWAIWy1v8DhtoH6U+HOxcAAAAASUVORK5CYII=" /><!-- --></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(oldpar)</span></code></pre></div>
<p>This gives us a general idea of what the potential viruses are.</p>
<ul>
<li>Plot (1) indicates that the mappability profile suggests there is
areas that can be mapped to other viruses or the host</li>
<li>Plot (2) indicates that the virus segment infected the sample</li>
</ul>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">5. References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-sugarbeet" class="csl-entry">
Candresse, T, A Marais-Colombel, and V Brault. 2022. <span>“<span class="nocase">Sugar beet sequencing data (RNAseq)</span>.”</span>
Recherche Data Gouv. <a href="https://doi.org/10.15454/MK1JIW">https://doi.org/10.15454/MK1JIW</a>.
</div>
<div id="ref-iimi" class="csl-entry">
Ning, H., I. Boyes, I. Numanagić, M. Rott, L. Xing, and X. Zhang. 2024.
<span>“Diagnostics of Viral Infections Using High-Throughput Genome
Sequencing Data.”</span>
</div>
<div id="ref-virhunter" class="csl-entry">
Sukhorukov, G, M Khalili, O Gascuel, T Candresse, A Marais-Colombel, and
M Nikolski. 2022. <span>“VirHunter: A Deep Learning-Based Method for
Detection of Novel RNA Viruses in Plant Sequencing Data.”</span>
<em>Front Bioinform</em> 2 (May). <a href="https://doi.org/10.3389/fbinf.2022.867111">https://doi.org/10.3389/fbinf.2022.867111</a>.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
