<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Haochen Ning" />


<title>Introduction to the iimi package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Introduction to the iimi package</h1>
<h4 class="author">Haochen Ning</h4>
<h4 class="date">First created on 2023-04-29. Updated on 2024-10-31</h4>


<div id="TOC">
<ul>
<li><a href="#introduction" id="toc-introduction">1. Introduction</a>
<ul>
<li><a href="#installation" id="toc-installation">1.1.
Installation</a></li>
<li><a href="#loading-packages-and-data" id="toc-loading-packages-and-data">1.2. Loading packages and
data</a></li>
<li><a href="#data-pre-processing" id="toc-data-pre-processing">1.3.
Data Pre-processing</a></li>
</ul></li>
<li><a href="#converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame" id="toc-converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame">2.
Converting BAM file(s) into coverage profiles and feature-extracted data
frame</a>
<ul>
<li><a href="#state-the-path-to-the-folder-of-your-bam-files" id="toc-state-the-path-to-the-folder-of-your-bam-files">2.1. State the
path to the folder of your BAM files</a></li>
<li><a href="#create-a-data-frame-that-contains-the-coverage-profiles." id="toc-create-a-data-frame-that-contains-the-coverage-profiles.">2.2.
Create a data frame that contains the coverage profiles.</a></li>
</ul></li>
<li><a href="#predicting-the-plant-samples" id="toc-predicting-the-plant-samples">3. Predicting the plant
sample(s)</a>
<ul>
<li><a href="#using-pre-trained-models-and-no-customization" id="toc-using-pre-trained-models-and-no-customization">3.1 Using
pre-trained models and no customization</a></li>
<li><a href="#customizing-your-own-model" id="toc-customizing-your-own-model">3.2. Customizing your own
model</a></li>
<li><a href="#customizing-unreliable-regions" id="toc-customizing-unreliable-regions">3.3. Customizing unreliable
regions</a></li>
</ul></li>
<li><a href="#visualizing-the-coverage-profiles" id="toc-visualizing-the-coverage-profiles">4. Visualizing the coverage
profiles</a></li>
<li><a href="#references" id="toc-references">5. References</a></li>
</ul>
</div>

<div id="introduction" class="section level2">
<h2>1. Introduction</h2>
<p>This vignette aims to give an introduction on how to use the
<code>iimi</code> package for plant virus diagnostics and how to
visualize the coverage profile for the sample mapping. We also included
a tutorial on creating unreliable regions.</p>
<div id="installation" class="section level3">
<h3>1.1. Installation</h3>
<p>First, let’s install necessary packages. You may skip this step if
you have installed the packages before.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install iimi</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">&quot;iimi&quot;</span>, <span class="st">&quot;dplyr&quot;</span>))</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co"># install Biostrings</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;Biostrings&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-packages-and-data" class="section level3">
<h3>1.2. Loading packages and data</h3>
<p>We will load necessary packages and data before we start any
analysis.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(iimi)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># this is the extracted DNAStringSet data from the Virtool virus database GitHub. It contains both version 1.4.0 and version 1.5.0 </span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>get_url <span class="ot">&lt;-</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;https://github.com/virtool/iimi/blob/main/data/virus_segments.rda?raw=true&quot;</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">url</span>(get_url))</span></code></pre></div>
</div>
<div id="data-pre-processing" class="section level3">
<h3>1.3. Data Pre-processing</h3>
<p>We provide three example coverage profiles to demonstrate how to use
<code>iimi</code>. These files are sourced from the dataset used in the
VirHunter paper <span class="citation">(Sukhorukov et al. 2022)</span>,
which we also utilized for external validation in our manuscript <span class="citation">(Ning et al. 2024)</span>. You can download these files
directly from the Recherche Data Gouv website <span class="citation">(Candresse, Marais-Colombel, and Brault 2022)</span>.
We recommend storing all BAM files in a single folder for ease of
access.</p>
<p>To get started creating coverage profiles and feature-extracted data
frames, the process involves four steps: (1) Downloading the raw data in
FASTQ format from the link above (2) Use Bowtie2 to map the FASTQ files
(paired-end or single-end) against the <a href="https://github.com/virtool/ref-plant-viruses">official Virtool
virus data base (ver. 1.4.0)</a> to get SAM outputs (3) Convert SAM to
indexed and sorted BAM using Samtools (4) Generate coverage profiles and
feature-extracted data frames using <code>iimi</code> functions
(tutorials in the next section)</p>
<p>We recommend Bowtie2 or minimap2 since we have tried both and they
yield similar result with minimap2 having a slight decrease. We let both
software to report all alignments (<code>-a</code> mode for Bowtie2,
<code>--secondary=yes</code> for minimap2). You can also use other
mapping tools.</p>
<p>Next, we provide a short tutorial to guide you through using
<code>iimi</code> functions to make predictions on the real data.</p>
</div>
</div>
<div id="converting-bam-files-into-coverage-profiles-and-feature-extracted-data-frame" class="section level2">
<h2>2. Converting BAM file(s) into coverage profiles and
feature-extracted data frame</h2>
<p>Let’s convert the indexed and sorted BAM file(s) into coverage
profiles and feature-extracted data frame from the previous section.</p>
<p>We will use the coverage profiles to visualize the mapping
information. The feature-extracted data frame will be used in the model
training and testing process.</p>
<p><strong>Note that both training and testing data need to go through
the conversion step.</strong></p>
<p>In our example, we stored the conversion for both the testing and
training datasets in the same object. You can do the conversion
separately for your data.</p>
<p><strong>Important: the example code does not work unless the path to
the folder that stores your BAM files is provided.</strong></p>
<div id="state-the-path-to-the-folder-of-your-bam-files" class="section level3">
<h3>2.1. State the path to the folder of your BAM files</h3>
<p>If you already have coverage profiles in run-length encoding (RLE)
format, go to section 2.2.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>path_to_bamfiles <span class="ot">&lt;-</span> <span class="fu">list.files</span>(</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">path =</span> <span class="st">&quot;path/to/your/BAM/files/folder&quot;</span>,</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">pattern =</span> <span class="st">&quot;bam$&quot;</span>,</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">full.names =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">include.dirs =</span> <span class="cn">TRUE</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="create-a-data-frame-that-contains-the-coverage-profiles." class="section level3">
<h3>2.2. Create a data frame that contains the coverage profiles.</h3>
<div id="convert-bam-files-to-a-list-of-rles." class="section level4">
<h4>2.2.1. Convert BAM files to a list of RLEs.</h4>
<p>You may skip this step if you already have converted them to RLE
format.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>example_cov <span class="ot">&lt;-</span> <span class="fu">convert_bam_to_rle</span>(<span class="at">bam_file =</span> <span class="st">&quot;path_to_bamfiles&quot;</span>)</span></code></pre></div>
</div>
<div id="convert-the-list-of-rles-to-a-feature-extracted-data-frame." class="section level4">
<h4>2.2.2. Convert the list of RLEs to a feature-extracted data
frame.</h4>
<p>This section explains how to convert the RLE format to
feature-extracted dataframes, with options for handling unreliable
regions.</p>
<p>By default, the package uses provided unreliable regions and enables
mappability profiling and filtering to eliminate false peaks. This is
the recommended approach for most users. * To use the default settings,
no additional code is required. * To disable this feature, set
<code>unreliable_region_enabled = FALSE</code> when calling
<code>convert_rle_to_df()</code>.</p>
<p>If you wish to use your own unreliable regions, refer to section 3.3
for instructions on creating custom unreliable regions and input your
custom unreliable regions using the <code>unreliable_region_df</code>
parameter in <code>convert_rle_to_df()</code>.</p>
<p><strong>Note: Customization of unreliable regions must be done before
calling <code>convert_rle_to_df()</code>. Most users can skip this step
and use the provided unreliable regions.</strong></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Using default settings (recommended)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">convert_rle_to_df</span>(example_cov)</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Disabling unreliable region processing</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">convert_rle_to_df</span>(example_cov, <span class="at">unreliable_region_enabled =</span> <span class="cn">FALSE</span>)</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Using custom unreliable regions</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Refer to section 3.3 for details</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>custom_regions <span class="ot">&lt;-</span> <span class="fu">create_custom_unreliable_regions</span>()</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">convert_rle_to_df</span>(example_cov, <span class="at">unreliable_region_df =</span> custom_regions)</span></code></pre></div>
</div>
</div>
</div>
<div id="predicting-the-plant-samples" class="section level2">
<h2>3. Predicting the plant sample(s)</h2>
<p>To make predictions, use the converted mapping result of the
sample(s) that you wish to detect as the input, <code>newdata</code>.
Make sure you have converted the indexed and sorted BAM files into
feature-extracted data frame from the section above.</p>
<p>After preparing your test sample, you can choose to test the data
using our provided training model or the model you trained using
<code>train_iimi()</code>. The tutorial of training your own model is
provided in the next section.</p>
<p><strong>Note: if you wish to customize unreliable regions, please go
to 3.3.</strong></p>
<div id="using-pre-trained-models-and-no-customization" class="section level3">
<h3>3.1 Using pre-trained models and no customization</h3>
<p>If you wish to use provided training model, only input your data to
<code>newdata</code> and choose a method of your wish using
<code>predict_iimi()</code>.</p>
<p>There are three methods that you may choose from: <code>xgb</code>,
<code>en</code>, and <code>rf</code>, which stand for pre-trained
XGBoost, elastic net, and random forest models. The example below uses
the pre-trained XGBoost model.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>prediction_default <span class="ot">&lt;-</span> <span class="fu">predict_iimi</span>(<span class="at">newdata =</span> df, <span class="at">method =</span> <span class="st">&quot;xgb&quot;</span>)</span></code></pre></div>
<p>The detection of your plant sample(s) is finished. The prediction is
<code>TRUE</code> if virus infected the sample, <code>FALSE</code> if
virus did not infect the sample.</p>
</div>
<div id="customizing-your-own-model" class="section level3">
<h3>3.2. Customizing your own model</h3>
<p>If you would like to train your own model and use this model to test
your data, you can use the codes below to train a new model with your
own data.</p>
<p>Ideally, the number of the samples used to train the model should be
bigger than 100. However, we are only providing a tutorial on how to use
the <code>train_iimi()</code> function, only two samples are used to
train the model since <code>example_cov()</code> only contains three
in-house data’s coverage information.</p>
<p>Now, we need to prepare our training data. We are using a 80/20
random split to split the three samples. This means that two samples are
used as the training data, and one sample is used as the testing data.
If you are training your own data, the training data is the data that
you want to train the model on; the testing data is the data that you
would like to have a prediction on.</p>
<p>Here are some definitions/explanation of the objects to input in
<code>train_iimi()</code>:</p>
<ol style="list-style-type: decimal">
<li><p><code>train_x</code>: the feature-extracted data frame of plant
samples that you would like to train <code>iimi</code> model on. Make
sure that you have mapped the samples to the virus database and
converted the mapping result to sorted and indexes BAM files.</p></li>
<li><p><code>train_y</code>: the known truth or labels for your
<code>train_x</code> data. Please label the data to make sure that it
has a detection label for virus segments as well.</p></li>
<li><p><code>test_x</code>: the feature-extracted data frame of plant
samples that you would like to predict using your trained
<code>iimi</code> model. Make sure that you have mapped the samples to
the virus database and converted the mapping result to sorted and
indexes BAM files.</p></li>
</ol>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># preparing the train and test data</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># spliting into 80-20 train and test data set with the three plant samples</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>train_names <span class="ot">&lt;-</span> <span class="fu">sample</span>(<span class="fu">levels</span>(<span class="fu">as.factor</span>(df<span class="sc">$</span>sample_id)),</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">length</span>(<span class="fu">unique</span>(df<span class="sc">$</span>sample_id)) <span class="sc">*</span> <span class="fl">0.8</span>)</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="co"># trian data</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>train_x <span class="ot">=</span> df[df<span class="sc">$</span>sample_id <span class="sc">%in%</span> train_names,]</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># test data</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>test_x <span class="ot">=</span> df[df<span class="sc">$</span>sample_id <span class="sc">%in%</span> train_names <span class="sc">==</span> F,]</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="co"># preparing labels</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>train_y <span class="ot">=</span> <span class="fu">c</span>()</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (ii <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(train_x)) {</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  train_y <span class="ot">=</span> <span class="fu">append</span>(train_y, example_diag[train_x<span class="sc">$</span>seg_id[ii],</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>                                         train_x<span class="sc">$</span>sample_id[ii]])</span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Then, we plug in the variables into the <code>train_iimi</code>
function with the default XGBoost model to train your custom model.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">train_iimi</span>(<span class="at">train_x =</span> train_x, <span class="at">train_y =</span> train_y)</span></code></pre></div>
<p>Now, we have a trained model using the toy data.</p>
<p>Then, the process to detect which viruses infect the plant sample(s)
is the same as previously described, except we are using a customized
trained model.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>prediction_customized <span class="ot">&lt;-</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">predict_iimi</span>(<span class="at">newdata =</span> test_x,</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">trained_model =</span> fit)</span></code></pre></div>
<p>The detection of the plant sample(s) is finished. The interpretation
is the same as above.</p>
</div>
<div id="customizing-unreliable-regions" class="section level3">
<h3>3.3. Customizing unreliable regions</h3>
<p><strong>Note: if you would like to create your own unreliable
regions, please customize them first, then extract features to build a
data frame from section 2.2.2. using customized unreliable
regions.</strong></p>
<p>If you would like to create your own unreliable regions besides from
using your own training model, you may use
<code>create_mappability_profile()</code> and
<code>high_nucleotide_regions()</code>. Both functions’ output is a data
frame with the start and end position of the unmappable region, the
virus that the region is on, and the category that it belongs to.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># An example of the provided unreliable regions</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>unreliable_regions <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Categories) <span class="sc">%&gt;%</span> <span class="fu">sample_n</span>(<span class="dv">2</span>)</span></code></pre></div>
<pre><code>## # A tibble: 8 × 6
## # Groups:   Categories [4]
##   Start   End `Virus segment` Categories                `1_4_0` `1_5_0`
##   &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;           &lt;fct&gt;                     &lt;lgl&gt;   &lt;lgl&gt;  
## 1 35360 35465 6as0dk9c        A% &gt; 45%                  TRUE    TRUE   
## 2    19   158 t4er406c        A% &gt; 45%                  TRUE    TRUE   
## 3    52   179 j3gc98g1        CG% &gt; 60%                 FALSE   TRUE   
## 4   819   944 xkwztnid        CG% &gt; 60%                 FALSE   TRUE   
## 5     1   394 937ob21p        Unmappable region (host)  TRUE    TRUE   
## 6  8837  8915 wrta20tr        Unmappable region (host)  FALSE   TRUE   
## 7  2772  2846 fe75665p        Unmappable region (virus) FALSE   TRUE   
## 8  2330  3124 v4t9nq3k        Unmappable region (virus) FALSE   TRUE</code></pre>
<p>Unreliable regions contains (1) mappability profile (2) high
nucleotide content regions.</p>
<ul>
<li><p>Mappability profile is a profile of areas on a virus genome that
can be mapped to (1) other viruses or (2) host genome. We choose
Arabidopsis Thaliana as our host genome.</p></li>
<li><p>High nucleotide content regions is a profile of areas on a virus
genome that has (1) high GC content and/or (2) high A nucleotide
percentage.</p></li>
</ul>
<p>Including these two profiles into <code>iimi</code> ensures that
there are no false peaks like the ones described in the previous
section.</p>
<div id="mappability-profile" class="section level4">
<h4>3.3.1. Mappability profile</h4>
<p>Here is a short tutorial to make mappability profile.</p>
<ol style="list-style-type: decimal">
<li><p>Split each of the virus segment from the virus database into a
sliding window series with window size of your choice and with step size
1. The default value for window size is 75. You may choose any window
size you want.</p></li>
<li><p>Map one virus segment with each other, until you finish mapping
it to all virus segments in the virus database. Also map the virus
segment with a host genome of your choice. We chose to use Arabidopsis
Thaliana.</p></li>
<li><p>Sort and index the resulted BAM files from the mapping
step.</p></li>
<li><p>Assemble the mappability profile:</p></li>
</ol>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># if you would like to keep unmappable regions that can be mapped to other</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># viruses or the host genome separate into two data frames, you may use the</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># following code:</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># input your own path that you would want to store regions on a virus that can</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co"># be mapped to another virus</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># you can customize the name of this type of mappability profile</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>mappability_profile_virus <span class="ot">&lt;-</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/virus&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region (virus)&quot;</span>, <span class="at">virus_info =</span> virus_segments)</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># input your own path that you would want to store regions on a virus that can</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a><span class="co"># be mapped to the host genome</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a><span class="co"># you can customize the name of this type of mappability profile</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>mappability_profile_host <span class="ot">&lt;-</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/host&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region (host)&quot;</span>, <span class="at">virus_info =</span> virus_segments)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="co"># if you would like to keep everything in the same data frame, you may use the</span></span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co"># following code:</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>mappability_profile <span class="ot">&lt;-</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_mappability_profile</span>(<span class="st">&quot;path/to/bam/files/folder/of/both/types/&quot;</span>, <span class="at">category =</span> <span class="st">&quot;Unmappable region&quot;</span>, <span class="at">virus_info =</span> virus_segments)</span></code></pre></div>
</div>
<div id="high-nucleotide-content-regions" class="section level4">
<h4>3.3.2. High nucleotide content regions</h4>
<p>Creating the high nucleotide content regions is much easier than the
mappability profile. We only need to use
<code>create_high_nucleotide_content()</code> function.</p>
<p>Here is an example:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>high_nucleotide_regions <span class="ot">&lt;-</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">create_high_nucleotide_content</span>(<span class="at">gc =</span> <span class="fl">0.6</span>, <span class="at">a =</span> <span class="fl">0.45</span>, <span class="at">virus_info =</span> virus_segments)</span></code></pre></div>
<p>The default threshold for GC content is 60% and is 45% for A%. The
thresholds are customizable.</p>
<p>Now, we can combine these two regions into the final unreliable
regions. You can use them to convert your training and testing data to
feature-extracted data frames. Please refer to section 2.2.2. to see how
to do so.</p>
</div>
</div>
</div>
<div id="visualizing-the-coverage-profiles" class="section level2">
<h2>4. Visualizing the coverage profiles</h2>
<p>Next, we can visualize the coverage profile by using the
<code>plot_cov()</code> function.</p>
<ul>
<li><code>plot_cov()</code>: plots the coverage profile of the plant
sample and the percentage of A nucleotides and GC content for a sliding
window of k-mer with the step as
<ol style="list-style-type: decimal">
<li>We used the default setting of k = 75.</li>
</ol></li>
</ul>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>oldpar <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">2</span>))</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot all segments of one sample, you can try:</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cov(covs = example_cov[&quot;S1&quot;])</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot all segments from all samples, you can try:</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_cov(covs = example_cov)</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a><span class="do">## if you wish to plot certain segments from one sample, you can try:</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>segs <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;82np2784&quot;</span>, <span class="st">&quot;m0kacxse&quot;</span>)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>covs_selected <span class="ot">=</span> <span class="fu">list</span>()</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>covs_selected<span class="sc">$</span><span class="st">`</span><span class="at">305S</span><span class="st">`</span> <span class="ot">&lt;-</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  example_cov<span class="sc">$</span><span class="st">`</span><span class="at">305S</span><span class="st">`</span>[segs]</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="do">## if you have many segments that you would want to plot, you can try the following code with the numbers changed</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="do">## to find the index of your desired segments:</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co"># covs_selected$`305S` &lt;-</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co">#   example_cov$`305S`[names(example_cov$`305S`)[c(8, 15)]]</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">4</span>, <span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="fu">layout</span>(<span class="fu">matrix</span>(<span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">4</span>), <span class="at">nrow =</span> <span class="dv">3</span>))</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_cov</span>(<span class="at">covs =</span> covs_selected, <span class="at">virus_info =</span> virus_segments)</span></code></pre></div>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAqAAAAHgCAMAAABNUi8GAAABEVBMVEUAAAAAADoAAGYAOjoAOmYAOpAAZmYAZpAAZrY6AAA6ADo6AGY6OgA6Ojo6OmY6OpA6ZmY6ZrY6kJA6kLY6kNtmAABmADpmAGZmOgBmOjpmOmZmOpBmZjpmZmZmZpBmZrZmkGZmkJBmkLZmkNtmtrZmtttmtv9paWmQOgCQOjqQOmaQZjqQZmaQZpCQkDqQkGaQkLaQkNuQtraQttuQtv+Q27aQ29uQ2/+2ZgC2Zjq2Zma2kDq2kGa2tma2tpC2tra2ttu22/+2/7a2/9u2///bkDrbkGbbkJDbtmbbtpDbtrbbttvb25Db27bb29vb/7bb/9vb////AAD/tmb/tpD/25D/27b/29v//7b//9v///9o9graAAAACXBIWXMAAA7DAAAOwwHHb6hkAAAgAElEQVR4nO2dC3vcxnWGh7KVkHZcNTUVJ3Zs1VZDyDHTtLFjUnbTxAkpK27ahtnQJfH/f0gxGMwVA+DM4AA7C37vI3F3sXPDwYu5YG+iBqBgxL4bAMAYEBQUDQQFRVOioN//+75bsDX6ET2YGK8l6OvHNy8v5J27n755dXtydHH3VJzWtydvXrVPXwtxdKHTXp/afNen8pG7xSf+jNx699Ojs36qtmrnGXn/Uohjub2+fesiLKlYZMOzIqq3eNuCBCpC8u+1+OjkzQt5+GSSv3f329sOWfXw8ZnNSoK++vDJze5Y3rt+9/LssolO++j6+PLs7r2rNkDXp3JXm/9fvpCuqPtfdIK+cXQmH+3Eo399Q/yitUnmuTxrCmq2NaWpbc2D35386Pj6DSGfkbe/UKW8PJP31YHRzzTHsE1V3396Ibfevzg6HEFlw5MjuhNH7/z2g6+F+MWLo5+fDodRFdb8/cHTox+d7U7l4ZPH4LK7L2/r+vKdow/F6WUb0ZdnuybOrazvvtT1ckRzrR709slN80/e2735TbN/x9eiOUnvPvuHb9qn2/Nd9mTyv1Ry9+bnMgyPv+560N2xfPT6WBrWJpJFvn501YayPVhNhibbcd0W3iTqtGvzfS1UxibT00ff6GeaA/jkprkvj5Hcfv3up4cjqNyJpIjKJDJWH/9Y7reMq9wSD6MsXP1t0pw0wWtqap/t7re3tTorVNXH9fc/E22J9y8e37RHrz1Is1ldULVPqhuzA01zozarp5qg//CzUy3oZXtuG0FVwibTyx+d1rqYY5mhE1Sq1wSqNU/l+/oH76tj0B4m/Yw8UH9t7svHzfYfvhDi8c1K8ZiNbHRSRDtBb9/5whc0FkbjZ9u5vj7Vgtr7r9sSTdVNfC+ff92WePfTR1fq6LEM/CsKumvjdNuNBPJWzQjbAak538/k/En+f92MDc1wexYZ4pWg3URrd6REawb2d45lBvVADvEymG3P0Q5aTSm7H6sANwPRmX5GlirvyxbI7d1xOQxkw5Mi2g7xzbjyyyu536/NEN8Poyq8ibzsf9syWinfODL35a1SX00K2viKd07V/XaydGZmw/NYcRX/cvGj/0qtfXaDQ8urM/fRglP7dciIqByAafvtxyr9eSZKvMwEgCFDUAECZh4CEDBX0HnHY3vMFZSnFdsBgjIDQXmBoMxAUF4gKDMQlBe6oLt2yhq+YoWABkBQXsiC3r9oL3V1L6iRsjxIICgvZEHbVyTMDSnLgwSC8pLcg74NQUeBoLzQ56C3J5iDEoCgvGAVzwwE5SVVUDsHjbwSldUAjkIKYi1BtxSzMfbegwqOQgoCgvICQZmBoLwkLZKOF7jMBEHzsm8pZmMkXWbaPbqCoBNAUF7SLtTfvfdHCDoOBOWF3oO2H9G5+xn3hXoImpd9SzEbI2UOKl9Kakb54fy5DdhSsFcSdFvX5kbAKp4ZCMoLBGUGgvICQZlZS1CZcktxGwKCMrOeoA+jE4WgzKwp6KYCNwAEZWYdQWXnCUGT8+c2YEtxXk/QrhfdOBCUmRUFZaiufCAoMysJylRd+UBQZiAoLxCUGQjKCwRlBoLyAkGZgaC8QFBmICgvEJQZCMoLBGVmZUE3FbsYEJQZ8r7M+rZACJqVP7cBWwoydV/mfVsgBM3Kn9uALQWZui/zvi0Qgmblz23AloKc2oPmfQgRgmblz23AloJM3pdZ3xYIQbPy5zZgS0HGKp4XCMoMVvG8QFBmVl/Fbyl4ESAoM2uv4jcVvAgQlJm1V/GbCl4ECMrM2qv4TQUvAgRlJntfkr5SHYJm5c9twJZinDIHvRbisb9GQg8aAkGZSRB0d8ywit9U8CJAUGYSBL0+xSp+EgjKDFnQp4//8ry5+QA96CgQlBn6vtyeHF3cvoVV/DgQlJm1X4vfVPAiQFBmICgvEJSZFQQVEDQzf24DthRj+iJJXZjP+FEKCJqbP7cBW4oxeV/ufnIR2QpBAyAoM/R92YWvIlGzQ9Dc/LkN2FKMl5+DBt/9vaXgRYCgzEBQXiAoMxCUFwjKDATlBYIys7igAoJm589twJZivLygNQTNzZ/bgC3FeAVBawiamT+3AVuK8dKCCgianz+3AVuK8eKC1hA0O39uA7YUYwjKCwRlZmFBhfoDQbPy5zZgSzGGoLyQBZ33MdnxBmwpxhCUlxRBZ3xMdrwBW4rxsoIKCDrEvI/JjjdgSzFeWNDuLwTtMe9jsuMN2FKMISgvCYukOR+THW/AlmK8iqDuVzht/AtCsYpnZnVBN25oqqB2Dpr0bWzjDdhShNcQtA4E3VL8QtCDMrOooPpZCJqXP7cBWwrw+oJue5CHoMxAUF4gKDP7EHRTAQxIuA6a/U0Ykw3YUnz3I+iWIuhD70HzvwljsgFbCu+SgloPe4JuKYQeCUN89jdhTDZgS9FdVNDBZBCUlD+3AVuK7jqC9p4R24qiBYIysx9B5cvz2+xFISgz+xKU41W9EoGgzOxJ0ODD8tsBgjKzoKBTl6C2FEYDBGVmb4JuK4wGCMoMBOUFgjIDQXmBoMwsJ+jkKn1LYTRAUGYWFHThqssEgjIDQXmBoMzsU9AtxVEDQZnZo6CbiqMGgjIDQXmBoMwsJihhAN9SHDUQlJnlBF287iKBoMxAUF4gKDMQlBcIygwE5QWCMgNBeYGgzBymoOUeAQjKzF4Fza284I+LQFBm9iloduUQdLyAYqOTAQTlBYIyc5iClvtGEwjKDATlBYIyc3iCtuM7BB0roNTg5EDel137XQtH5B+lWFbQYo8BBGWGui/3L87kDf2H0ZYWtNCDAEGZoe5L92sU9B9Gg6AQlIPUHvTtQgQt1VAIygx5X25PipqDlnoUICgzB7iKn5N5cSAoMwe4ip+TeXEgKDMHuIrXt0UeBgjKzMGt4gd/maEMICgzh7aKd/rNIg8DBGXm0FbxEJRQQJGRySR7X6Z+PXopQWflXh4IysyhreLdLCUukyAoM3tdxWdE0he0vCMBQZnZ6yp+pqAlHgkIysxeV/EQdDx/bgPKC0s+S63iaeXOFbS8QwFBmZm5LxA0AIIyA0F5gaDMQFBeICgz5FX8U3Vh/hFxkQRBwwcZ+XMbUFxUZkDel7ufXES2bkRQviMKQZmh78vu8U1/41YEZXuBCoIygzloW8weBE197TihARB0OvvhChp9AZUsLVnQ5NeOExoAQaezLyRoKArPoRAEQYk1kQVNfu2YCAQlZc85nDkZWN4uIoJ3ma4iaPJrx0QgKCn7WoJyHAv/u54GBM3ao9E8qe8AJwJBSdkPSlCvnNUEJeTPAYKSsq8mKMPB2Kugdg469REFWvXd9wJtR1EIShU0Y9m3fg8qgtvDB4L2BY3XAkH3AgT1BfVtddPQxl8IygwEpQpKm9lBUGaWETTrqmFehkMVNPXtYTQgKCl73guDeRlYrstMCap3aLoyeg+a+PYwGhCUlJ1c6IEIKpYQNPHtYTQgKCn74Qrq22qSCPKBxxyUGQg6LaiAoPtjCUHpZXIIOvdYEAStIejegKCuk/Gvv4ege2QBQROUYRB0tqGi9xeCFsQSgi5Y+8wTYrjIyFreTQFB98UmBGX4FNGEoKMNGEoLQRngFzSlRBZBZxrqDusQtDi2IGhOQb2sELRMIGhU0GHPJmqCoMywC5o02pYlqDNT8D8on1ATBGWGX9DFapf+8AsqIGjRHI6gYhFB3Suc7oePh+ed41VBUGYOSFD5j19Qe8dzEoIWwsEIqhYxawnqvTsZgu6RAxOUoaCBjIH7ELQQIKi+ExYBQYvgsATlKGgoIwQtEm5B0150LEnQwWeGzZ0qD4LOh13QpWpfRtCRfBC0CCDo5FMQdJ9A0MmnIOg+gaCTT0HQfQJBx56KXtuCoGsCQUeeEhB07xyIoFPvmV9C0O6zciNX8KcKhKDz2Yigufsxlk19kqRfMwRdk4MRlKskejbVhUZOjdFMMxsFQQMORVC2kpJyxT+OB0FXBIKmp4CgKwJBM3JRrp5mFg9BA8oRVIzlL0pQ0uX9zOIhaMB+BXWTq08crSroMldPZ5UPQQOKEVToRTOtnpn1zsg0nhGCMsMsaGpxjqC1qAd+ao3yM28Q1MsOQQeyzxG0+2CxCBOQvnppXUGJn96DoPMpS9C6Dn97WKieNbkhibWzZYWgzJQkqLlzCIJSVnMQdD4FCipEKGhGQxJrZ8sLQZkpRVDva5FE3NaUhqTVnpOXcLkBgs6nGEHdbfYLk/J+Bju5dr7MEJQZ8p7s2s7sKPj5viUEtY8iq/rJgrJq58sMQZmh7sn9izN5c/vE//m+hQXNKCir9qzM0dwQlBnqnty9d+XcDGbPFVRENy8lqFDpWX4eZGwjBJ1Pag/69qig6XER0XF8JUFT8hCrhKDMkPfk9oQwB80QVEQX6sqfhQTVVULQA4D3IOUJGhlrlxRUQNADgncVnyPoYLOSfv2IntS8K4XnNxTHNkLQ+fCu4nPmoIObF/q9EHuNFYIeANmr+GgnxCsoQzmRhKJ/Lw8Iugb7XsUPsKCgyW1JKgmCMrPvVfwAIrWDOzhB5Vh0LcRjf8oEQUP2vYofLjetrEMUdHc8/dJcZpsenqC0E55R0MQZ4iEKen06/dJcZpsepKCEE55T0NQM3AnzSqIL+vTxX543Nx+gBx0lQVDCCQ9BExZJtydHF7dvTbw9LLNND1BQ0gnPJ2hqSQcoKL3Q9OwPT1DaCf+gBI0WlSroxIXl3CY9REEp2fcoKDED55HjEJRSZkZ2CBrPvj9BqVVD0AMDguYzT1DaZ2gymwRB49n3GJdDE5T4GZrMJkHQePZ9xoVWdzGCEj9Dk9kkCBrPDkGTe9Dxd9/kNgmCRrPvNSwHJijt3Te5TYKg0ewQFKt4biAob40QlBkIylsjBGUGgvLWCEGZgaC8NUJQZiAob40QlBkIylsjBGUGgvLWCEGZeWCCzn2/5WSNEJSZhybo0jVCUGa2Iyipdgh6aGxJUEL1EPTQeBCCisg9jgqntkHQ+WxI0MHqxUKCxkqDoMw8DEG1oswNhKDL8wAEFcJ8FygEPTg2L2j3HSXqK7+5G9gvD4IysyVB46sW0QlK/tnPWRVCUGa2Lqj9ro75X9tBqRCCMrN5QVeuEIIys3FB+TvNqQrHn84oMSv7vo8EH1sUtLVykWX7UI0DGyDofDYoqF0PLdx/1hB0ebYnqOo/1U/JQdCDZ3OCLrpqj9Y4vAGCzodT0L1HJfjZTwi6ATYm6Bq9plfj6AYIOp9NCbpKp+nXN7oBgs5nY4KuDQRdGgg6Cwi6NBB0FhB0aSDoLCDo0kDQWUDQpYGgs4CgSwNBZwFBlwaCzgKCLg0EnQUEXRoIOgsIujQQdBYQdGkg6Cwg6NJA0FlA0KWBoLOAoEsDQWcBQZcGgs4Cgi4NBJ1HuM8QlBkIOg8IujAQdB4QdGEg6Dwg6MJA0HlA0IWBoPOAoAvDKOh2gpIABF0YCDqP8IP4EJQZCDqTbEF37fegHF2MFpfbmu0cCwg6k1xB71+cyZvbJzdjxeW2ZjvHAoLOJFfQu/eunBtSloTWbOdYQNCZBF9XltyDvg1BR4GgsxGDQRiNyO0J5qAEyHsyPaffTlASyRN0uqwZTdnOsaDuyeScfu1v5iyIOYLaOagQ/tebiixmZS6IlBi6gRyJJ7GgDTIYhvV70O2R2oPyzum3BwRlhhyQReb022PPF+q3B+ebRcDeL9RvDwjKy54v1G8PCMrLni/Ubw8IysueL9RvDwjKC1bxzEBQXiAoMxCUl9mCgoCZxwMEzBQ0L3NC2r0nXarYNctaorx9NBCCrljsmmUtUR4ELSQpBF2pQAialxSCrlQgBM1LCkFXKhCC5iWFoCsVuLSgACwNBAVFA0FB0UBQUDQQFBQNBAVFA0FB0UBQUDQQFBTNDEF3QpxNpWk/cKcSun/73L8Q4piWtL7sJRppyvUpNWlT7KMrWtpremPpMBTR0d+TWZAPYkp51DbmC3r3k4vw43T9NE+bRqiE7t9Iyutj+Rk9UtLdsdxFUlL5sapTWgPq+08vzF5NpL19ckNtLB2GIjr6ezKvYeSDmFAeuY35gsqj9OnFaJL7X/6xOVlUQvfvQPLrU2rSRlBa0vtPf35KbICKESmt7JWJSekwFKFL6u3JnNISDyK1PGob8wVtOrL6cqqnl725Suj+HUxLTHrdjAi0pNenjUy0pLs3Tppxm5T28mNyUjoMReiSensyr7yUg0guj9rGcgRtOnp6GPxEg0mbc5MsaP1/N/U1rdjLxzfUFtDhE7S/J/OKW0JQchsXHeJVW4gj7FsXCaMmcTZwLT+CRZ44kIuV0aTPR4jwDfESv3nzyqIfRHp55DYuu0hq20Jbd7RbqYukRhHqrP2aukiSxV6Tl2kJLSDCt0jq78m88sgHMaE8chtLucx02fZ15MtM5GtHaZeZqGmvU1pAhfUy0yljw5a6zERqIy7Ug6KBoKBoICgoGggKigaCgqKBoKBoICgoGggKigaCgqKBoKBoICgoGggKigaCgqKBoKBo8CsfDMw8BCBgrqDzjsf2wO8k8QJBmYGgvEBQZiAoLxCUGQjKS76gkTksmGEY4hkFPSgz6EF5gaDMQFBeuAT9m4va9Oqivn0+nPPbK/W/vzWooX9R7P784/PnzZ/3gw/4//mb/r1hqsq/lfz9yc1gekqZbIIinrq5gw9odFn+NhTQ73790VXz/3cfXX1/3t67kf/l/W//+aObb6/uz581u/Dd+ZfN7ffnH3/+3c3deZtGBzZ21bYpuf6v5s/9H+r6rsnYZfnTy3f/2jxS94YD01F1gazcgH71O3mMZBNlw747/23z996U+d/ncjc+/lw1ejQguSCe8YBEHtAYydIG9NXF7RevLnZnt1/8z82fm3vtluf6/rdXX335l+cyqbz96qrdcqVyjVT6lTw651f6/jc6iypG3UvflYa7Z3fP2pbfPlcltaWbMmWzZV2q0ckBoYB4jgWEWdCzWgbo9otvr/7etOfrT/6zuXf3TP6397/6ze8/lwNRc/ubX9006f/0qxu5fTSgMlxf/fqivvs3GdD7v+osTQCa4tS99F2RBX/y+/N2pLx7pkpqSr+xZTbN/lw2UjU6OSAUEM+xgPAKevf+h+9f2YD+x798dvrqn84v5H99/9ur3bPPzuQeyNvdP/78tEn7idw+GtD78w8/fPa/as4kk+ssux+/bopR9yaHpFi5v7qpd7IDapq2a0uSJe9Mmb9umi0bqRqdHBAKiOdYQBZfxccn8po/X71m+j6imQw3cbqRq67iH1o8cZlpPrjMxAsEZQaC8gJBmYGgvEBQZiAoLxCUGQjKCwRlBoLyAkGZgaC80AW9PTn67VP5i3XD+UFCQBBPEmRB738pf4ChvvvAf0kBAQ2gBgTxpEEWVP4ipvwFkvf8Ux4BDaAGBPGkkdqD3j7BGT9KYg+KeE5An4PuBOZMBMgBQTxJYBXPDFbxvKQKaudM+BRilMSAIJ4ToAdlBj0oLxCUGQjKS8oiSXIUvPcfAQ1IWCQhngTol5letG98xmWRCciXmRBPEikX6p0bUpYHScKFeucmOfuDIbkHfRsBHSW1B0U8x0l5swjmTAQS3iyCeBLAKp4ZrOJ5gaDMQFBeICgzEJQXCMoMBOUFgjIDQXmBoMxAUF4gKDMQlBcIygwE5QWCMgNBeYGgzEBQXiAoMxCUFwjKDATlBYIyA0F5gaDMQFBeICgzEJSXfEHxMdko2QFBPKOgB2UGPSgvEJQZCMoLBGUGgvJSmKA5xZU1b4OgvEBQZiAoL6UJmtOEjEwico8HCDofMXC/CEH7BYqJSiDotvAvtm1B0IxGQNByKVNQVU70SrWzrX+n7gvaK2KsSPZL4xB0Lv7rFUsLSi2yEzQyCRWuTeaPlzUUNCgDgu6l8tzICu+AHqaggk/QEq5D8GWfV3chgg4/4hK0IhUZqXpcUHVvStD+NMGb0ogujXk4GUdKVNjmtNPZ2/1Pz0apm9PQ+BsLJioQ/TSLCyr8ZIOdt25bsqCCIKgI7hot44KK4ASYZGjOlM7kBQvRO0Hr/uOJOrpiTAjsdkojvKSDz/eOC6Fk1aoiBRVjgtY2biRBwzLGBI0e3NIFHR0jpqsw+HlN8AmlEQSto3OryZb1Dh49e7zMyLbKCqpP+Ligwk3XPduXyxO0f4TCk65L7ymsyxUjgoYZRncxeHYlQa1ZYTcT61b94kTtBsFR1BXVn/+MtMnkHWxoJKPoBSpIFnvDYZqg958GX2c5IKgx1ImECNP7lunFebet6lK3uoWC+tchRFCU56w+BCag+sZq3O82zDELm9mjf6CSBKXF01ZmTwevRvVEdM6no2OOgdZbb7SFiq4sEe5NZKwWTqHRhnq3uinhDvjd18DgMPgg4O6p2p1HlG8EpggaBGJY0DoiaLAqmRC0DgWtddS5BHW2DCf2SIpnr4WBoEa9SPscQY0Owj5tpNFlOU/XbtCCVlIEDWdxto82HbZt91BZgw9Cbk9OqT1oRFDvmDsNNsUEgtrcNtO0oLoMEWwxBdkDYOPVj88agibE09QVTWi6ISeg3pMmYl5Pap92O4Fgf3SnFunWguPqN9TmdzM4B6MTXB+qMdkHH/S5PvqSEFChBQ2rDw63Hxin228fV7IQx6taP+t0BrYFpnSTKiKoPl/rakBQfR7YzLarjROcQfGADEKLp2mJCBPquu25ZwdL01U5ERSRpbXw0ptgmAfa7aBimzYmqCt4cMfrsoQ9BQb7gRRB6/sX4VeqRwWt+4KaU3NQULPjrqBueF1B3XNzUtCw61SCtsVUcUFtf1KTBHUTJAhKi6duSbQT07vkZzdKubaZWBCKdyMh/BA4wa/84iP7EBU0OK1Hes9e8SnBHcoiHEGdvtO/sU/bU9YIahrsnP9apClBzTzLbugLqqt0BXUb5gkadiDBzs4SNFbiUEU9QW0vFOhT1zaCXpFjgvqG6N3vCVq5nV1VOYmj+2DXYasJOvnjp0IP8WYcdgXw+ndn6BVtp2klE82jyh0WjJKORc6doIN2hzYrd6UENdcYbF/q9rlW0K6R5vhU7gVe3ZiZgo7E032gm+pWJPpi1W3gnECH1YU7MNDs3uDdFVW1R6W7a1aybt4qHKvNoe6V79W8Wg8qI9S2c0BQR1G9VR9hV9AuFO4OhfvtN0BfP3JG9mlBK3+mJmyJpiOaFDTsl/h6UOdIC+/isn46UpsMnD2Hg2erEUGnyu6slPrbU6UK2twtHOLFjW4Yacrgg4z8agiQm9oTzRlmtTtjggozN2wFrccF7eaow4Lqc9cRVPQErbrSxwXVQ1p33vl7v6SgplNv1QhSDghaDwtaxU6yeDP0eOY2pvNTX+ZudVT1mTHfO2p+U0c3jLRk8EEI6cdP9Qggd8QMysIMBe4orzbpPqp2elBtl/+ek15NMr3fy7QPK+eBFbQ9+3U3oHO4grZF2umRGimNoEId2sobVE3RWYJOxlPoM6dSI3woaLzYSowaSBa0Db+btolHpU9UFf7urzvWKUHDc6k3PYm0a6hhZEGJP35adUO8I6jps8xw63R0g4JWVeW+ZEoVNHyRoHuiUoI6LWzrEEZQVaSaoVSqfqFH9ylBveNBFXQ6niYaFd36qj3tRp8nDvMRQStdgt2u++Tu/K7Vwe4LOllnVQ2cO2RBp3/8tBOz2w9XUDObrqKC6u7UEbSqSIKKvqD++wCq2giqX4G1g6XupIWpUlS+oGo8ELqkrkpb2wxBJ+NpxphK75Z3/OKHvD3HBuaaVRcBLdC4NJWwwqhcnWVhS7qgCiuo96wyb6KurkGxZMk96PCPn6pD2O6HmsJoeSorqCOKzW0EtSemCrQjmzkJTdDUkfMXEsIdw4RTghqJKvPAJFCPlaDqyoEKp9BDmJ2m6BmWqLsbRyFaDF0m42nG08oeQDcCanT1JPKaGuJsIgnqeFW5i/agrK4plXAvhdhQD4/dXlmVTtxrOX0OOvnjp21FanLdXiVyp84qbaXPRZ1DH/3uUeWW5a9GTL9nzv8qLqgNnnBOZm99adujGyhMdOQ2Jag+4WxZalbRDgi+oFlz0Kl4Ooe48tS0glYxQXWmoLqek1MzAXuuR+YFvYf6aAg/gfGaIGm84XRB4zgB1YJWuvOyglb6jnNTB4K6l5VUGc6e9QSt+oK254ZzDvqCqgxeDLpuuV1/VAY1I+guQbll6Q5VD/pG0CpL0Di97O054w0E9og7Ezf/fHKGC1OO+eNusFm9J/RcwPRsU82uROULaoqu9OEaKcN/zk/IKGjdTZbMFaZuQmhOc0dUr0fQ4jhLwdpOCFQ17qDRpW6nPU6naNaVOrDC5leXrfQxNEWJ7lHVjVLdsNnuhLr4qKfR+gJjVXfbhRKzCiddfIJ2TXOv2zgBdMNRxSTyD3rggNa10o5XbkZnrqW7g8mGV4GckX4zXkiv5X5lnIK2x7Obp8WWfLWZg/qC+uerE83KtFbo/s9OxIQOr95adaOzjbAT6rZP7h0JfZBUx1R1L4K1hQn/sHuCdosllV9UWXPQAUJBI9qF693K9lJ+0uF+qTJedyH0eriw8DB3jCqYk0UG67jl/Y1LC6omaMIOP04fVnuC2rm/FTJooR2NHUGVQ8ZGnVVUwjdf16fv9S4l1s5B6qTUvbA/w1CFq0vV7Ziu3sjSXbnPm4PGCbP3Be3PLu2sZ6jXrGIF1TY0lRP0wH5S9xkRtJ8vVsxAp+ps5hO0VlO0bu4palc701XqAdTc1v2TtPLua0PNWrlzSPWg3qCnh3dv/LNjoK3D6ywqI6guMh5u/Qpsd9FHqDOkt2JmFTR2Qo3oMtBlRfLovsH0GTYz3aNJJhSt+r1+Px+roLUraLfFOQkrcwHELkqcDsA2zxdM1dNdZdfDul7DWGvNq3BOD21q1oJ654KtvMtgGl2pq6BuRIV6IarqVoDSV64+icwAAAPTSURBVCV20HExDvHjKwuX0WTRUiJjlxlpUisYzDE85zAj6EDb7FY+QbvDVruXcypHCTUqO72p7fScOac/UplJpl75OH2kcNMqQe2kwQqqe4de2eaI2FC6Vx67SYod41tBhToPRHcq6p2ZF0OXvOyjvSphEa4FrUaLYmhT0E0PVFYtJqj7UqNxqXbOFbdFlU5krp0YrdyWOidb5WQylfjlOf/1SVx5KoUx6SumjpPQ/bHepq4sGUGds8fLvxdBB3DbT0jt/OWofGhjtyQlnTjLCmrHPz2WW/3MMFxrQW23GeyP3uRYHs4Bqk49cxJ4d4Iu2avC6WxtpXqRZPUzF88q1bsaQStvr+qyBK0jc+TR1LQF0SwqsyhdW1D9djanKY4bkWuhzvaeW04R7lzM5nKqCPLXfoqgF6mCv97swi3UfVNO1c1b9Du1dI12hIgGJAdeQWNDxkja5fV0OihiBk5Be+/y82aXvgq9/tIdlP1CrKDOyO4+bwuowo3eOGy3O38HA6Un0l0vbAS13bi/A362XJgFTfGTcXgfq6I/KxplQUHd4JgxvP+6gUkwhJV7VFA/j11dBb1jaOWYoLZzrrq3GAgzOemXZLLNYeuC1oUK6s49o+XQmjuUKrLdnXj2kjpuEQRVpYm6tiO8+0Qv2xy4h/iDh/M66Gj2WOfXezqYz+kc4/1ANS5oNDN9PWBK0YIGZoelQFBeVhd09OmoN9OCRp+NLH+8J6njmV4Hmx7Uzh1iQFBe8gV1PlrERrwznHBpYNYwmon+Ik2tL6QG5Q5lzw7IEvHcAJw96HwGjvpU7xtfJ41XRO5Co9MO9KBrUZagA0yqlL78pI/xQy/HQNBVKE7QjEsdOVdHUq5JJ5UPQXkpS9CBzmo616JA0D1SmqBZvSFrG2YWD0F5KUvQlV7MSAOC7pPCBF3j/QrLAkF5gaDMQFBeChP08IGgvEBQZiAoLxCUGQjKCwRlBoLyAkGZgaC8QFBmICgvEJQZCMoLBGUGgvJCF5T0IwqAHBDEkwRZUOKPKDx4qAFBPGmQBZ3+EQUgoQYE8aSR3IMO/4gCkKT2oIjnOPQ56OSPKAAJOSCIJwms4pnBKp6XVEEnf+34oZMYEMRzAvSgzKAH5QWCMgNBecGFemZwoZ6X2RfqQQAx8IgnEaqgAxeWQSaIZwYZF+pBJohnBhkX6kEuiGc6mKGDooGgoGggKCgaCAqKBoKCooGgoGggKCgaCAqKBoKCooGgoGggKCgaCAqKBoKCooGgoGggKCgaCAqKBoKCooGgoGggKCgaCAqKBoKCooGgoGj+H8uRGSkmqruxAAAAAElFTkSuQmCC" /><!-- --></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(oldpar)</span></code></pre></div>
<p>This gives us a general idea of what the potential viruses are.</p>
<ul>
<li>Plot (1) indicates that the virus segment did not infect the
sample</li>
<li>Plot (2) indicates that the virus segment infected the sample</li>
</ul>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">5. References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-sugarbeet" class="csl-entry">
Candresse, T, A Marais-Colombel, and V Brault. 2022. <span>“<span class="nocase">Sugar beet sequencing data (RNAseq)</span>.”</span>
Recherche Data Gouv. <a href="https://doi.org/10.15454/MK1JIW">https://doi.org/10.15454/MK1JIW</a>.
</div>
<div id="ref-iimi" class="csl-entry">
Ning, H., I. Boyes, I. Numanagić, M. Rott, L. Xing, and X. Zhang. 2024.
<span>“Diagnostics of Viral Infections Using High-Throughput Genome
Sequencing Data.”</span> <em>Briefings in Bioinformatics</em>. <a href="https://doi.org/10.1093/bib/bbae501">https://doi.org/10.1093/bib/bbae501</a>.
</div>
<div id="ref-virhunter" class="csl-entry">
Sukhorukov, G, M Khalili, O Gascuel, T Candresse, A Marais-Colombel, and
M Nikolski. 2022. <span>“VirHunter: A Deep Learning-Based Method for
Detection of Novel RNA Viruses in Plant Sequencing Data.”</span>
<em>Front Bioinform</em> 2 (May). <a href="https://doi.org/10.3389/fbinf.2022.867111">https://doi.org/10.3389/fbinf.2022.867111</a>.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
